includes/classes/class-plugin-manager.php

```diff
@@ -0,0 +1,409 @@
+								esc_url( $status['url'] ),
+								/* translators: %s: Plugin name and version. */
+								esc_attr( sprintf( _x( 'Install %s now', 'plugin', 'default' ), $name ) ),
+								esc_attr( $name ),
+								__( 'Install Now', 'default' )
+							);
+						} else {
+							$action_links[] = sprintf(
+								'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+								_x( 'Cannot Install', 'plugin', 'default' )
+							);
+						}
+					}
+					break;
+
+				case 'update_available':
+					if ( $status['url'] ) {
+						if ( $compatible_php && $compatible_wp ) {
+							$action_links[] = sprintf(
+								'<a class="button aria-button-if-js" data-plugin="%s" data-slug="%s" href="%s" aria-label="%s" data-name="%s">%s</a>',
+								esc_attr( $status['file'] ),
+								esc_attr( $plugin['slug'] ),
+								esc_url( $status['url'] ),
+								/* translators: %s: Plugin name and version. */
+								esc_attr( sprintf( _x( 'Update %s now', 'plugin', 'default' ), $name ) ),
+								esc_attr( $name ),
+								__( 'Update Now', 'default' )
+							);
+						} else {
+							$action_links[] = sprintf(
+								'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+								_x( 'Cannot Update', 'plugin', 'default' )
+							);
+						}
+					}
+					break;
+
+				case 'latest_installed':
+				case 'newer_installed':
+					if ( is_plugin_active( $status['file'] ) ) {
+						$action_links[] = sprintf(
+							'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+							_x( 'Active', 'plugin', 'default' )
+						);
+						if ( current_user_can( 'deactivate_plugin', $status['file'] ) ) {
+							global $page;
+							$s       = isset( $_REQUEST['s'] ) ? $_REQUEST['s'] : ''; // phpcs:ignore
+							$context = $status['status'];
+
+							$action_links[] = sprintf(
+								'<a href="%s" id="deactivate-%s" aria-label="%s" style="color:red;text-decoration: underline;">%s</a>',
+								wp_nonce_url( 'plugins.php?wpp=1&action=deactivate&amp;plugin=' . rawurlencode( $status['file'] ) . '&amp;plugin_status=' . $context . '&amp;paged=' . $page . '&amp;s=' . $s, 'deactivate-plugin_' . $status['file'] ),
+								esc_attr( $plugin['slug'] ),
+								/* translators: %s: Plugin name. */
+								esc_attr( sprintf( _x( 'Deactivate %s', 'plugin', 'default' ), $plugin['slug'] ) ),
+								__( 'Deactivate', 'default' )
+							);
+						}
+					} elseif ( current_user_can( 'activate_plugin', $status['file'] ) ) {
+						if ( $compatible_php && $compatible_wp ) {
+							$button_text = __( 'Activate', 'default' );
+							/* translators: %s: Plugin name. */
+							$button_label = _x( 'Activate %s', 'plugin', 'default' );
+							$activate_url = add_query_arg(
+								array(
+									'_wpnonce' => wp_create_nonce( 'activate-plugin_' . $status['file'] ),
+									'action'   => 'activate',
+									'plugin'   => $status['file'],
+									'wpp'      => 1,
```

Comment:
Related to my above comment on the PHP handler, this would be better as the following, to make use of the `admin_action_{$action}` hook:
```suggestion
									'action'   => 'perflab_activate_plugin',
									'plugin'   => $status['file'],
```